ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -369,87 +373,162 @@ func (scbuilder *SiteConfigBuilder) getWorkloadManifest(cpuSet string, role stri
+		if err != nil {
+			return nil, nil, err
 		}
+		dirFiles = &DirContainFiles{
+			Directory: path,
+			Files:     files,
+		}
+		dirFilesArray = append(dirFilesArray, *dirFiles)
 
-		filePath := scbuilder.scBuilderExtraManifestPath + "/" + file.Name()
-		if strings.HasSuffix(file.Name(), ".tmpl") {
-			// For templates, we can inject the roles directly
-			// Assumes that templates that don't care about roles take precautions that they will be called per role.
-			for role := range roles {
-				filename, value, err := scbuilder.getManifestFromTemplate(filePath, role, clusterSpec)
-				if err != nil {
-					return dataMap, err
-				}
-				if value != "" {
-					value, err = addZTPAnnotationToManifest(value)
+	case *[]string:
+		for _, p := range *path {
+			files, err = GetFiles(p)
+			if err != nil {
+				return nil, nil, err
+			}
+			dirFiles = &DirContainFiles{
+				Directory: p,
+				Files:     files,
+			}
+			dirFilesArray = append(dirFilesArray, *dirFiles)
+
+		}
+
+	}
+
+	for _, v := range dirFilesArray {
+		for _, file := range v.Files {
+			if file.IsDir() || file.Name()[0] == '.' {
+				continue
+			}
+
+			filePath := v.Directory + "/" + file.Name()
+
+			if strings.HasSuffix(file.Name(), ".tmpl") {
+				// For templates, we can inject the roles directly
+				// Assumes that templates that don't care about roles take precautions that they will be called per role.
+				for role := range roles {
+					filename, value, err := scbuilder.getManifestFromTemplate(filePath, role, clusterSpec)
 					if err != nil {
-						return dataMap, err
+						return dataMap, doNotMerge, err
+					}
+					if value != "" {
+						value, err = addZTPAnnotationToManifest(value)
+						if err != nil {
+							return dataMap, doNotMerge, err
+						}
+						dataMap[filename] = value
+						// Exclude all templated MCs since they are installation-only MCs
+						doNotMerge[filename] = true
 					}
-					dataMap[filename] = value
-					// Exclude all templated MCs since they are installation-only MCs
-					doNotMerge[filename] = true
 				}
-			}
-		} else {
-			// This is a pure passthrough, assuming any static files for both 'master' and 'worker' have their contents set up properly.
-			manifestFile, err := ReadExtraManifestResourceFile(filePath)
-			if err != nil {
-				return dataMap, err
-			}
+			} else {
+				// This is a pure passthrough, assuming any static files for both 'master' and 'worker' have their contents set up properly.
+				manifestFile, err := ReadExtraManifestResourceFile(filePath)
+				if err != nil {
+					return dataMap, doNotMerge, err
+				}
 
-			manifestFileStr, err := addZTPAnnotationToManifest(string(manifestFile))
-			if err != nil {
-				return dataMap, err
+				manifestFileStr, err := addZTPAnnotationToManifest(string(manifestFile))
+				if err != nil {
+					return dataMap, doNotMerge, err
+				}
+				dataMap[file.Name()] = manifestFileStr
+
+				// assuming user provided custom CRs are not in the first directory path
```

Comment:
this comments is not clear to me. If it is not needed just remove it.

Reply:
deleted with commit 4e71357842c233ccb388973c901535f4c82919cd